aip/cloud/2510.md

```diff
@@ -22,30 +22,33 @@ different services use different project identifiers.
 ## Guidance
 
 **TL;DR:** The project number is the canonical identifier, and the project ID
-is an [alias][], and behaves as such. Additionally, third-party services are
-unable to _accept_ project IDs.
+is an [alias][]; however, unlike normal aliases, it **may** be returned if it
```

Comment:
This guidance is really confusing.  Why for APIs are we not saying Project ID is the *canonical identifier*?  (other documentation says internally we can only store and log Project Numbers ) Its what everyone uses.   I feel like this is like  a surpreme court decision trying to not override a precedent but carve out an exception.

Thoughts?

Reply:
It's not entirely clear to me what "third party APIs are forced to use the project number" means here. For example, I can create an API which accepts GCP resource names from a call and pass them on to a GCP service - I'm not forced to use a number for that.

Could you clarify how this restriction is manifested in reality?

Reply:
What I actually meant was third-party _services_ running on our infra and using projects, which are barred from using project IDs at all for privacy reasons.